ddtrace/contrib/internal/ray/span_manager.py

```diff
@@ -0,0 +1,220 @@
+from contextlib import contextmanager
+from itertools import chain
+import threading
+import time
+
+from ddtrace import config
+from ddtrace import tracer
+from ddtrace._trace.span import Span
+from ddtrace.constants import SPAN_KIND
+from ddtrace.ext import SpanKind
+
+from .utils import _inject_ray_span_tags
+
+
+@contextmanager
+def long_running_ray_span(span_name, service, span_type, child_of=None, activate=True):
+    """Context manager that handles Ray span creation and long-running span lifecycle"""
+    span = tracer.start_span(name=span_name, service=service, span_type=span_type, child_of=child_of, activate=activate)
+    span.set_tag_str(SPAN_KIND, SpanKind.CONSUMER)
+    _inject_ray_span_tags(span)
+    start_long_running_span(span)
```

Comment:
We should wrap this whole method in a contextmanager (with tracer.start_span(...): ....). This will ensure that the span is finished if `_inject_ray_span_tags` or `start_long_running_span` raise an error.